scripts/zkai_native_seq32_attention_mlp_dry_run_opening_sampler_gate.py

```
@@ -0,0 +1,862 @@
+            }
+        )
+    return output.getvalue()
+
+
+def require_output_path(path: pathlib.Path) -> pathlib.Path:
+    target = pathlib.Path(os.path.abspath(path if path.is_absolute() else ROOT / path))
+    evidence = EVIDENCE_DIR.resolve()
+    try:
+        relative = target.relative_to(evidence)
+    except ValueError as err:
+        raise DryRunOpeningSamplerGateError("output path must stay under evidence directory") from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)